src/components/IndexMapSwitch.js

```diff
@@ -0,0 +1,63 @@
+import PropTypes from 'prop-types';
+import React, { useContext } from 'react';
+import { StyleSheet, View } from 'react-native';
+
+import { texts } from '../config';
+import { SettingsContext } from '../SettingsProvider';
+
+import { Button } from './Button';
+
+export const IndexMapSwitch = ({ filter, setFilter }) => {
+  const { globalSettings } = useContext(SettingsContext);
+  const { navigation = 'drawer' } = globalSettings;
+
+  return (
+    <View style={[styles.floatingButtonContainer, stylesWithProps({ navigation }).position]}>
+      <Button
+        onPress={() => {
+          const selectedFilter = filter.find((entry) => entry.selected);
+          const updatedFilter = filter.map((entry) => {
+            if (entry.id !== selectedFilter.id) {
+              entry.selected = true;
+            } else {
+              entry.selected = false;
+            }
+
+            return entry;
+          });
+
+          setFilter(updatedFilter);
+        }}
+        title={
+          filter.find((entry) => entry.selected).title === texts.locationOverview.list
+            ? texts.locationOverview.map
+            : texts.locationOverview.list
+        }
+        notFullWidth
+      />
+    </View>
+  );
+};
+
+const styles = StyleSheet.create({
+  floatingButtonContainer: {
+    alignSelf: 'center',
+    position: 'absolute'
+  }
+});
+
+/* eslint-disable react-native/no-unused-styles */
+/* this works properly, we do not want that warning */
+const stylesWithProps = ({ navigation }) => {
+  return StyleSheet.create({
+    position: {
+      bottom: navigation === 'tab' ? 0 : '5%'
+    }
+  });
+};
```

Comment:
Wouldn't it be more correct to use `drawer` instead of `tab` here? Because there is a possibility that the navigation value is `undefined` in `globalSettings`. But we use `drawer` as standard. If the navigation value is not used in `globalSettings`, the navigation structure of the application will be tabbar. What kind of view will we get in this case?

|globalSettings.navigation = drawer|globalSettings.navigation = tab|globalSettings.navigation = undefined|
|--|--|--|
![Simulator Screenshot - iPhone 14 - 2023-08-21 at 16 24 20](https://github.com/smart-village-solutions/smart-village-app-app/assets/11755668/8bf16ccb-5a76-4232-a369-6dd095ce76c0) | ![Simulator Screenshot - iPhone 14 - 2023-08-21 at 16 24 50](https://github.com/smart-village-solutions/smart-village-app-app/assets/11755668/dc095384-f7ff-45b4-8719-af67dc3c940d) | ![Simulator Screenshot - iPhone 14 - 2023-08-21 at 16 25 10](https://github.com/smart-village-solutions/smart-village-app-app/assets/11755668/695e0710-331f-48d5-8d6b-83caebdd9959)

Reply:
i guess there is a development ongoing to change the default navigation to tabs. when we are not defining a navigation, the fallback value needs to be aligned with the resulting navigation. so in your third case the value should result in `tab`, which would result in a correct positioning with `0`. so we do not need to change the code here, or am i wrong?

Reply:
I'm a little confused about this. Looks like you're right. If we don't specify a navigation, we will automatically have tab navigation. I can't say anything as the development is not finished yet. I will merge the two branches in my local and share the results with you. 👍

Reply:
The result looks the same as the screenshots above. The reason is that we set `navigation` as `drawer` in line 12 of `IndexMapSwitch`.

|globalSettings.navigation = drawer|globalSettings.navigation = tab|globalSettings.navigation = undefined|
|--|--|--|
![Simulator Screenshot - iPhone 14 Pro Max - 2023-09-14 at 12 00 12](https://github.com/smart-village-solutions/smart-village-app-app/assets/11755668/2779ae1c-c337-41e9-ba6a-220d8c00e6dd) | ![Simulator Screenshot - iPhone 14 Pro Max - 2023-09-14 at 12 02 26](https://github.com/smart-village-solutions/smart-village-app-app/assets/11755668/0e380694-3588-47f4-8a20-572dcfefffa1) | ![Simulator Screenshot - iPhone 14 Pro Max - 2023-09-14 at 12 02 48](https://github.com/smart-village-solutions/smart-village-app-app/assets/11755668/be592754-76a7-42d7-a1ca-58e022d438ba)

how about adding a control like this?
```
return StyleSheet.create({
    position: {
      bottom: !navigation || navigation === 'tab' ? 0 : '5%'
    }
  });
```

Reply:
`!navigation` is never the case, look at your logs. it is always drawer or tab. that's why i mean to not change anything. if we correctly fallback to tab in case nothing is defined in global settings, your third case would result in `{"navigation":"tab"}` and the condition works.

Reply:
If we don't define anything for navigation in `globalSettings`, `navigation = 'drawer'`. Because in line 12 of the code, that is here:
```
const { navigation = 'drawer' } = globalSettings; 
```
we say that navigation is `drawer`.
Therefore the warning text in the screenshots says `{ 'navigation': 'drawer' }`. But nothing is set in `globalSettings`.
I guess we can't agree on that. 😂

Reply:
correct, which needs to be changed to fallback to tabs with our change of tab navigation as default.